cpu/atxmega/periph/timer.c

```diff
@@ -0,0 +1,493 @@
+
+    TC0_t *dev;
+    uint8_t pre;
+    uint8_t ch;
+
+    assert(TIMER_CH_MAX_NUMOF * TIMER_NUMOF < 32);
+
+    /* make sure given device is valid */
+    if (tim >= TIMER_NUMOF) {
+        return -1;
+    }
+
+    /* figure out if freq is applicable */
+    for (pre = 0; pre < PRESCALE_NUMOF; pre++) {
+        if ((CLOCK_CORECLOCK >> prescalers[pre]) == freq) {
+            break;
+        }
+    }
+    if (pre == PRESCALE_NUMOF) {
+        DEBUG("timer.c: prescaling failed!\n");
+        return -1;
+    }
+
+    dev = timer_config[tim].dev;
+
+    /* stop and reset timer */
+    dev->CTRLA = 0;                    /* Stop */
+    dev->CTRLFSET = TC_CMD_RESET_gc;   /* Force Reset */
+
+    /* save interrupt context and timer Prescaler */
+    ctx[tim].cb = cb;
+    ctx[tim].arg = arg;
+    ctx[tim].prescaler = (0x07 & (pre + 1));
+
+    /* Check enabled channels */
+    ctx[tim].channels = 0;
+    for (ch = 0; ch < 4; ch++) {
+        if (timer_config[tim].int_lvl[ch] != CPU_INT_LVL_OFF) {
+            ctx[tim].channels++;
+        }
+    }
+
+    if (timer_config[tim].type != TC_TYPE_0
+        && timer_config[tim].type != TC_TYPE_4) {
+        if (ctx[tim].channels > 2) {
+            DEBUG("timer.c: wrong number of channels. max value is 2.\n");
+            return -1;
+        }
+    }
+
+    if (timer_config[tim].type == TC_TYPE_2
+        || timer_config[tim].type == TC_TYPE_4
+        || timer_config[tim].type == TC_TYPE_5) {
+        DEBUG("timer.c: Timer version %d is current not supported.\n",
+            timer_config[tim].type);
+        return -1;
+    }
+
+    /* Normal Counter with rollover */
+    dev->CTRLB = TC_WGMODE_NORMAL_gc;
+
+    /* Compare or Capture Interrupt Level */
+    dev->INTCTRLB  = (timer_config[tim].int_lvl[0] << TC0_CCAINTLVL_gp);
+    dev->INTCTRLB |= (timer_config[tim].int_lvl[1] << TC0_CCBINTLVL_gp);
+    if (ctx[tim].channels > 2) {
+        dev->INTCTRLB |= (timer_config[tim].int_lvl[2] << TC0_CCCINTLVL_gp);
+        dev->INTCTRLB |= (timer_config[tim].int_lvl[3] << TC0_CCDINTLVL_gp);
+    }
+
+    /* Free running counter */
+    dev->PER = 0xFFFF;
+
+    DEBUG("timer.c: prescaler set to %d \n", ctx[tim].prescaler);
+    dev->CTRLA = ctx[tim].prescaler;
+
+    return 0;
+}
+
+int timer_set_absolute(tim_t tim, int channel, unsigned int value)
+{
+    TC0_t *dev;
+
+    if (channel >= ctx[tim].channels) {
+        return -1;
+    }
+
+    DEBUG("Setting timer %i channel %i to %04x\n", tim, channel, value);
+
+    set_oneshot(tim, channel);
+
+    dev = timer_config[tim].dev;
+
+    /* Compare or Capture Disable */
+    dev->CTRLB &= ~(1 << (channel + TC0_CCAEN_bp));
+
+    /* Clear Interrupt Flag */
+    dev->INTFLAGS &= ~(1 << (channel + TC0_CCAIF_bp));
+
+    /* set value to compare with rollover */
+    *(((uint16_t *)(&dev->CCA)) + channel) = (dev->CNT + (uint16_t)value);
+
+    /* Compare or Capture Enable */
+    dev->CTRLB |= (1 << (channel + TC0_CCAEN_bp));
+
+    return 1;
```

Comment:
```suggestion
    return 0;
```

See https://api.riot-os.org/group__drivers__periph__timer.html#ga2b440fa3b789669204f659de9238d752